### PR TITLE
SPI fixes and Temperature Monitoring Completion

### PR DESCRIPTION
**Pull request by ClarkPoon**
The [datasheet for the LMT87](https://www.ti.com/lit/ds/symlink/lmt87.pdf?ts=1633208579405&ref_url=https%253A%252F%252Fwww.google.com%252F) helpfully gives us a lookup table for calculating temperatures from voltages. I decided to change our lookup table to be based of of TI's. I also wrote a short test plan for temperature, but I did not make any significant changes to the test code

### CODE REVIEW

File: Config/Inc/config.h [hunk 1]
```diff
@@ -20,6 +20,7 @@ typedef enum SafetyStatus_e {SAFE = 0, DANGER = 1, OVERVOLTAGE = 2, UNDERVOLTAGE
 
 //--------------------------------------------------------------------------------
 // Basic Parameters of BPS layout
+
```
Comment (ClarkPoon) [inline Config/Inc/config.h:23]:
```suggestion
```

File: Apps/Src/Temperature.c [hunk 2]
```diff
@@ -141,15 +129,17 @@ ErrorStatus Temperature_ChannelConfig(uint8_t tempChannel) {
 		Minions[board].com.tx_data[5] = (0xF << 4) + AUX_I2C_NACK_STOP;
     }
 
-    // Send data
+	// Send data
     wakeup_sleep(NUM_MINIONS);
     LTC6811_wrcomm(NUM_MINIONS, Minions);
-    LTC6811_stcomm();
+	pec_error |= LTC6811_rdcomm(NUM_MINIONS, Minions);
```
Comment (ClarkPoon) [inline Apps/Src/Temperature.c:135]: Don't we have a version of this function that automatically checks the PEC error and retries a few times if it fails? Can we use that?

File: Apps/Src/Temperature.c [hunk 3]
```diff
@@ -141,10 +128,11 @@ ErrorStatus Temperature_ChannelConfig(uint8_t tempChannel) {
 		Minions[board].com.tx_data[5] = (0xF << 4) + AUX_I2C_NACK_STOP;
     }
 
-    // Send data
+	// Send data
     wakeup_sleep(NUM_MINIONS);
     LTC6811_wrcomm(NUM_MINIONS, Minions);
-    LTC6811_stcomm();
+	LTC6811_rdcomm(NUM_MINIONS, Minions);
```
Comment (ClarkPoon) [inline Apps/Src/Temperature.c:134]:
```suggestion
	LTC6811_rdcomm_safe(NUM_MINIONS, Minions);
```

File: Apps/Src/Temperature.c [hunk 4]
```diff
@@ -158,8 +146,13 @@ ErrorStatus Temperature_ChannelConfig(uint8_t tempChannel) {
  * @param mV from ADC
  * @return temperature in Celsius (Fixed Point with .001 resolution) 
  */
-int milliVoltToCelsius(uint32_t milliVolt){
-	return voltToTemp[milliVolt];
+int32_t milliVoltToCelsius(uint32_t milliVolt){
+	if (milliVolt < 3301) {
```
Comment (chaseblock) [inline Apps/Src/Temperature.c:150]: Magic numbers are bad

Comment (chaseblock) [reply Apps/Src/Temperature.c:150]:
```suggestion
	if (milliVolt < sizeof(voltToTemp)/sizeof(voltToTemp[0])) {
```

File: Apps/Src/VoltageToTemp.c [hunk 5]
```diff
@@ -2,32 +2,57 @@
...
-float frac = numerator/denom;			
-float retVal = frac + 30;	
-*/
+// Lookup table from TI's datasheeet can be downloaded here:
+// https://www.ti.com/lit/zip/snic009
 
 // Values generated by:
 /*
-from math import sqrt
+import pandas as pd
```
Comment (chaseblock) [inline Apps/Src/VoltageToTemp.c:10]: No

**Review by chaseblock — CHANGES_REQUESTED** [review]

**Review by chaseblock — APPROVED** [review]

**Pull request merged**